cobbler/settings/migrations/V3_4_0.py

```diff
@@ -4,231 +4,148 @@
+        Optional("authn_pam_service"): str,
+        Optional("autoinstall_snippets_dir"): str,
+        Optional("autoinstall_templates_dir"): str,
+        Optional("bind_chroot_path"): str,
+        Optional("bind_zonefile_path"): str,
+        Optional("bind_master"): str,
+        Optional("boot_loader_conf_template_dir"): str,
+        Optional("bootloaders_dir"): str,
+        Optional("bootloaders_formats"): dict,
+        Optional("bootloaders_modules"): list,
+        Optional("bootloaders_shim_folder"): str,
+        Optional("bootloaders_shim_file"): str,
+        Optional("bootloaders_ipxe_folder"): str,
+        Optional("syslinux_dir"): str,
+        Optional("syslinux_memdisk_folder"): str,
+        Optional("syslinux_pxelinux_folder"): str,
+        Optional("grub2_mod_dir"): str,
+        Optional("grubconfig_dir"): str,
+        Optional("build_reporting_enabled"): bool,
+        Optional("build_reporting_email"): [str],
+        Optional("build_reporting_ignorelist"): [str],
+        Optional("build_reporting_sender"): str,
+        Optional("build_reporting_smtp_server"): str,
+        Optional("build_reporting_subject"): str,
+        Optional("buildisodir"): str,
+        Optional("cheetah_import_whitelist"): [str],
+        Optional("client_use_https"): bool,
+        Optional("client_use_localhost"): bool,
+        Optional("cobbler_master"): str,
+        Optional("convert_server_to_ip"): bool,
+        Optional("createrepo_flags"): str,
+        Optional("autoinstall"): str,
+        Optional("default_name_servers"): [str],
+        Optional("default_name_servers_search"): [str],
+        Optional("default_ownership"): [str],
+        Optional("default_password_crypted"): str,
+        Optional("default_template_type"): str,
+        Optional("default_virt_bridge"): str,
+        Optional("default_virt_disk_driver"): str,
+        Optional("default_virt_file_size"): int,
+        Optional("default_virt_ram"): int,
+        Optional("default_virt_type"): str,
+        Optional("enable_ipxe"): bool,
+        Optional("enable_menu"): bool,
+        Optional("http_port"): int,
+        Optional("iso_template_dir"): str,
+        Optional("jinja2_includedir"): str,
+        Optional("kernel_options"): dict,
+        Optional("ldap_anonymous_bind"): bool,
+        Optional("ldap_base_dn"): str,
+        Optional("ldap_port"): int,
+        Optional("ldap_search_bind_dn"): str,
+        Optional("ldap_search_passwd"): str,
+        Optional("ldap_search_prefix"): str,
+        Optional("ldap_server"): str,
+        Optional("ldap_tls"): bool,
+        Optional("ldap_tls_cacertdir"): str,
+        Optional("ldap_tls_cacertfile"): str,
+        Optional("ldap_tls_certfile"): str,
+        Optional("ldap_tls_keyfile"): str,
+        Optional("ldap_tls_reqcert"): str,
+        Optional("ldap_tls_cipher_suite"): str,
+        Optional("bind_manage_ipmi"): bool,
         # TODO: Remove following line
```

Comment:
Is this left in on purpose?

Reply:
What do you mean left? We changed this to the keys only (so we removed the defaults), so the schema can detect if the keys supplied in the file are valid or not.

Reply:
Is the comment "# TODO: Remove following line" left in the file on purpose? If yes, what's stopping us from doing this task?

Reply:
We have multiple occurrences of the setting. At the moment if my memory serves me right we don't have an option for settings that are not saved to disk. We would need to find a way to enable that so we can remove the property from the schema and thus subsequently from the `settings.yaml` file.